src/qt/transactiondesc.cpp

```diff
@@ -91,9 +91,8 @@ QString TransactionDesc::toHTML(CWallet *wallet, CWalletTx &wtx, TransactionReco
         if (nNet > 0)
         {
             // Credit
-            if (CBitcoinAddress(rec->address).IsValid())
-            {
-                CTxDestination address = CBitcoinAddress(rec->address).Get();
+            if (IsValidDestination(rec->address)) {
+                CTxDestination address = DecodeDestination(rec->address);
```

Comment:
This incur in a double decoding since above `IsValidDestination` also decodes. Maybe swap:
```cpp
CTxDestination address = DecodeDestination(rec->address);
if (IsValidDestination(address)) {
```